### ISSUE TITLE
D-BUS Configuration

### ISSUE THREAD

**Issue by dj-curls** [Igalia/cog#245]
Hi, 

I'm wondering what I need to do to get cogctl to work over the D-BUS. 

I am getting the following errors 

````
root@buildroot:~ # XDG_RUNTIME_DIR=/tmp DBUS_SYSTEM_BUS_ADDRESS=unix:path=/run/d
bus/system_bus_socket cogctl -y open https://google.com
GDBus.Error:org.freedesktop.DBus.Error.NameHasNoOwner: Name "com.igalia.Cog" does not exist
root@buildroot:~ # XDG_RUNTIME_DIR=/tmp DBUS_SYSTEM_BUS_ADDRESS=unix:path=/run/d
bus/system_bus_socket cogctl open https://google.com

(cogctl:246): GLib-GIO-CRITICAL **: 14:20:53.552: g_dbus_connection_call_sync_internal: assertion 'G_IS_DBUS_CONNECTION (connection)' failed
Cannot autolaunch D-Bus without X11 $DISPLAY
````

Cog is started as a systemd unit:

````
[Unit]
Description=Cog Browser
After=weston.service
[Service]
User=root
BusName=com.igalia.Cog
TTYPath=/dev/tty7
Environment="COG_PLATFORM_FDO_VIEW_FULLSCREEN=1"
Environment="XDG_RUNTIME_DIR=/tmp"
Environment="DBUS_SYSTEM_BUS_ADDRESS=unix:path=/run/dbus/system_bus_socket"
ExecStartPre=/bin/sleep 2
ExecStart=/bin/cog --platform=fdo http://127.0.0.1/info
[Install]
WantedBy=multi-user.target
````
I've created a D-BUS service file like so 
````
[D-BUS Service]
Name=com.igalia.Cog
Exec="/bin/cog" 
User=root
SystemdService=cog.service
````
Are there any pointers or documents on how to get the D-BUS control working?

Thank you

**Comment by tobias-grasse** [comment]
Did you build cog with the cmake flag `-DCOG_DBUS_SYSTEM_BUS=ON`? Otherwise, cog will only be available on the session bus.

See https://github.com/Igalia/cog/blob/master/CMakeLists.txt#L53

**Comment by dj-curls** [comment]
Ah I missed that flag, I've now updated my cog.mk file to include that. 

Cleared out the buildroot/output/package/cog directory, and rebuilt. 

Unfortunately I still get the same error messages as above.

**Comment by aperezdc** [comment]
@dj-curls Could you provide the logged output for Cog? In particular, I would like to know if you see this line in the output of `journalctl -u cog` (assuming that you named your service unit `cog.service`):

```
Acquired D-Bus well-known name com.igalia.Cog
```

If there any reason why the control interface cannot be exposed to the system bus, there should be a warning message logged that starts with the following string:

```
Cannot expose remote control interface to system bus: …
```

Knowing the contents of the logs would be useful to help diagnose this :mag: 

Finally, make sure that the `com.igalia.Cog.conf` file generated by CMake (from the template at `dbus/policy.conf.in`) is installed into the `/etc/dbus-1/system.d` directory.

**Comment by asweet-thegoodpenguin** [comment]
I'm experiencing the same issue, the problem is that com.igalia.Cog.conf is being installed into /usr/etc/dbus-1/system.d/ when built through buildroot. If this file is moved into the correct location post-install then cogctl works as expected against the system bus

**Comment by Bastian-Krause** [comment]
@asweet-thegoodpenguin please see #296

**Comment by aperezdc** [comment]
I suppose we can close this issue after having merged #296, the next release will include the fix by @Bastian-Krause (thanks!)
